addons/sale_loyalty/models/sale_order.py

```diff
@@ -281,7 +281,7 @@ def _get_reward_values_discount(self, reward, coupon, **kwargs):
         discountable = 0
         discountable_per_tax = defaultdict(int)
         reward_applies_on = reward.discount_applicability
-        sequence = max(self.order_line.filtered(lambda x: not x.is_reward_line).mapped('sequence')) + 1
+        sequence = max(self.order_line.filtered(lambda x: not x.is_reward_line).mapped('sequence'), default=0) + 1
```

Comment:
The default sequence for `sale.order.line` is 10 so we should set it to something higher than that (as usually we want discount lines at the end of the order)
```suggestion
        sequence = max(self.order_line.filtered(lambda x: not x.is_reward_line).mapped('sequence'), default=10) + 1
```